src/test/java/ClientConnectionTest.java

```diff
@@ -2,52 +2,49 @@
 import org.junit.Assert;
 import org.junit.Test;
 
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.OutputStream;
+import java.io.*;
 import java.net.Socket;
+import java.util.regex.Pattern;
 
 class FakeSocket extends Socket{
     public String input = "";
+    public FakeOutputStream fos;
+
+    public FakeSocket(){
+        this.fos = new FakeOutputStream();
```

Comment:
I think you can use a `ByteArrayOutputStream` instead of maintaining a `FakeOutputStream`